api/types.go

```diff
@@ -464,6 +464,7 @@ type PinOptions struct {
 	Name                 string            `json:"name" codec:"n,omitempty"`
 	ShardSize            uint64            `json:"shard_size" codec:"s,omitempty"`
 	UserAllocations      []peer.ID         `json:"user_allocations" codec:"ua,omitempty"`
+	Expire               int64             `json:"expire" codec:"e,omitempty"`
```

Comment:
Too many problem in using time.Time

changed expire to int64 from time.Time
`time.Now()` has a monotonic clock reading, but `time.Unix(0, nano)` doesn't
have it. This creates problems while equating two times. Have faced some
more problems. It is safer to just use int64 instead.

If I keep time.Time, there will be back and forth conversion from time.Time to int64(string) which converting from pin options to query and back. The time that you get back with `time.Unix(0, nano)` is not the same. 
Plus with default value is simpler with int64, for time.Time  `time.Unix(0, 0)` is undefined

Reply:
Please use `time.Time`. We need to represent a date. It's the same on why we don't carry a []byte slice instead of peer.ID etc.

We don't have to use monotonic clocks. I don't even understand why you would need nanosecond accuracy for this.